bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -797,8 +838,9 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 		}
 		r.Log.Info("bootstrap data secret for KubeadmConfig already exists", "secret", secret.Name, "KubeadmConfig", scope.Config.Name)
 	}
```

Comment:
As an aside, if we don't rely on Status.Ready in `Reconcile()`, then we can likely remove the handling for an existing Secret here.

Reply:
added to the issue